src/test/java/org/json/junit/JSONMLTest.java

```diff
@@ -762,8 +762,7 @@ public void testToJSONObject_reversibility() {
         final String xml = JSONML.toString(originalObject);
         final JSONObject revertedObject = JSONML.toJSONObject(xml, false);
         final String newJson = revertedObject.toString();
-        assertTrue("JSON Objects are not similar",originalObject.similar(revertedObject));
-        assertEquals("original JSON does not equal the new JSON",originalJson, newJson);
```

Comment:
Instead of deleting line 766, replace it with code that converts the strings to JSONObjects, and compares them using JSONObject.similar().